### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by cmd-esc**
Added a section on instantiated objects vs object literals.  Added a section on current limitations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +20,8 @@ npm install --save git+https://github.com/StediInc/TypeSuite.git
 
 Or add it to your `package.json` manually.
 
+Don't miss the section on Current Limitations towards the bottom of this file.
```
Comment (asciimike) [inline README.md:23]: I think I would move this to the "currently in alpha" and link to it.

e.g. **Note: TypeSuite is currently in alpha and is subject to [several limitations](#current-limitations)...**

File: README.md [hunk 2]
```diff
@@ -184,6 +187,89 @@ const purchaseOrder = poResponse.readResponse.record as PurchaseOrder;
...
 for more!
 
+### A note on instantiated objects vs object literals
+
+Building requests to the NetSuite SuiteTalk API frequently involves instantiating quite a few objects.  In order to
+minimize the amount of code required to build all the necessary objects, in many cases an object literal can be
+provided instead of a fully instantiated object.  There are two cases where a fully instantiated object is
+required.  The first case is the outermost object, typically the Request object, which must be created by calling
+the class' constructor. In the example below, a GetRequest is explicitly instantiated via `new`.
+
```
Comment (asciimike) [inline README.md:197]: `GetRequest`

File: README.md [hunk 3]
```diff
@@ -184,6 +187,89 @@ const purchaseOrder = poResponse.readResponse.record as PurchaseOrder;
...
+```ts
+const getRequest = new GetRequest({
+  ...
+  });
+```
+
+The second case where a fully instantiated object is required is where you provide an object that is a subclass of
+type that is expected for that field.  For example, GetRequest has single field called `baseRef` whose type is
```
Comment (asciimike) [inline README.md:205]: Same

File: README.md [hunk 4]
```diff
@@ -184,6 +187,89 @@ const purchaseOrder = poResponse.readResponse.record as PurchaseOrder;
...
+
+As mentioned above, TypeSuite only supports the [2019_2 WSDL](https://webservices.netsuite.com/wsdl/v2019_2_0/netsuite.wsdl).
+
+### Available ports
+
+TODO: Quick implementation or write up
+
+### Date handling
+
+Right now, you must convert Dates to a string manually.  This is due to a complication with one of
```
Comment (asciimike) [inline README.md:250]: `Date`s

File: README.md [hunk 5]
```diff
@@ -184,6 +187,89 @@ const purchaseOrder = poResponse.readResponse.record as PurchaseOrder;
...
+the underlying libraries that TypeSuite uses and will be fixed in a later release.  However, it will always be
+possible to provide a string for any field in the NetSuite API that holds a Date.
```
Comment (asciimike) [inline README.md:252]: Same.

Would be good to have an example of what the date format is (`YYYY-MM-DDThh:mm:ss.sssZ`?) as well

File: README.md (hunk 1 above)
Comment (cmd-esc) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (cmd-esc) [reply README.md:196]: Done.

File: README.md (hunk 3 above)
Comment (cmd-esc) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (cmd-esc) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (cmd-esc) [reply README.md]: Formatting change done.

Comment (cmd-esc) [reply README.md]: As for an example of the date format, there's already an example of creating and using a DateFormatter in our example code.  Plus, any formatting requirements come from the WSDL, not from anything we are doing.

**Comment by cmd-esc** [comment]
Merged manually.

**Pull request closed without merging**